brion/spikeReport.cpp

```diff
@@ -36,36 +38,37 @@ const std::string spikePluginDSONamePattern( "Brion.*SpikeReport" );
 
 namespace brion
 {
+namespace
+{
+using SpikePluginFactory = lunchbox::PluginFactory<SpikeReportPlugin>;
+void _loadPlugins()
+{
+    static bool loaded = false;
+    if( !loaded )
+        SpikePluginFactory::getInstance().load( BRION_VERSION_ABI,
+                                                lunchbox::getLibraryPaths(),
+                                                spikePluginDSONamePattern );
+    loaded = true;
+}
+}
+
 namespace detail
 {
 class SpikeReport
 {
 public:
-    typedef lunchbox::PluginFactory< SpikeReportPlugin > SpikePluginFactory;
-
-    class DSOPluginsLoader
-    {
-    public:
-        DSOPluginsLoader()
-        {
-            SpikePluginFactory::getInstance().load( BRION_VERSION_ABI,
-                                                    lunchbox::getLibraryPaths(),
-                                                    spikePluginDSONamePattern );
-        }
-    };
-
     explicit SpikeReport( const SpikeReportInitData& initData )
     {
-        static DSOPluginsLoader loader;
-        plugin.reset( SpikePluginFactory::getInstance().create( initData ));
+        _loadPlugins();
```

Comment:
I think the previous code was thread-safe with regard to multiple threads creating the first SpikeReport at the same time because static variable initialization is thread safe (but I'm not 100% sure). The new version is not, or it it?